doc/new-guidelines.md

```diff
@@ -0,0 +1,468 @@
+and we are in the process of integrating a tool to format non-trivial code
+according to the Clojure Style Guide. This should greatly reduce noise in
+pull-request reviews and it'll simplify the life of all contributors.
+
+Pay special attention to:
+
+- Align let bindings https://github.com/bbatsov/clojure-style-guide#bindings-alignment
+- Align map keys https://github.com/bbatsov/clojure-style-guide#map-keys-alignment
+
+## Dos and don'ts
+
+### Component styles
+
+Prefer to define styles in separate files named `style.cljs`, usually colocated
+with the source file using it. For a real example, see
+[src/status_im/ui2/screens/chat/messages/style.cljs](../src/status_im/ui2/screens/chat/messages/style.cljs).
+Define styles as functions when you need to parameterize something, but prefer
+to define them as simple maps for a performance edge.
+
+```clojure
+;; bad
+(defn animated-checkbox-view [{:keys [size]}]
+  [animated/view
+   {:style {:width           size
+            :height          size
+            :border-radius   4
+            :justify-content :center
+            :align-items     :center}}
+   [animated/view
+    (do-something)]])
+
+;; good
+(defn animated-checkbox-view [{:keys [size]}]
+  [animated/view
+   {:style (styles/animated-checkbox-style size)}
+   [animated/view
+    (do-something)]])
+```
+
+### Using TODOs comments
+
+_TODO_ comments are used extensively in the codebase, but in order to easily
+grep them or be able to act on them, we have found the following recommendations
+to be useful:
+
+- Prefer to use TODO comments only when an issue alone is not enough. We don't
+  want to turn the codebase into a Project Management System 💣
+- If adding a TODO comment, consider adding a link to one or more issues.
+- If you think it's valuable, add a timestamp in ISO-8601 format (`YYYY-MM-DD`).
+  Remember that it's easy to move code around and lose track of the original
+  commit date that introduced the comment.
+- If you think it's valuable, add the Github username, so that people know who's
+  responsible for it. Most of the time this is not needed, because the team is
+  sharing the responsibility.
+
+```clojure
+;; bad
+;; TODO Refactor the world.
+
+;; good
+;; Use a colon to separate the description.
+;; TODO: Refactor the world.
+```
+
+```clojure
+;; bad
+;; We're a multicultural team, prefer ISO-8601.
+;; TODO(11-09-2022): Refactor the world.
+
+;; good
+;; TODO(2022-11-09): Refactor the world.
+```
+
+```clojure
+;; bad
+;; TODO(2022-11-09): @someone Refactor the world.
+
+;; good
+;; Use something people can more easily grep.
+;; TODO(2022-11-09,@someone): Refactor the world.
+```
+
+### Subscription names and event names
+
+Always register events and subscriptions using a meaningful namespace, but don't
+fully qualify them. We understand it's a controversial decision because there
+are both pros and cons to use FQK (fully-qualified keywords).
```

Comment:
I'm new to the term 'fully-qualified keywords' in clojure - I'm used to hearing them called 'namespaced keywords'.

Reply:
It seems this is coming from my inner Clojure circles, but it's not well known outside of them. Qualifying a keyword means adding a namespace, which is a synonym for namespacing a keyword. The problem comes that sometimes we want to refer to qualifications that are valid and are expanded to the current namespace (e.g. `::whatever`).

Maybe there's no consensus on the community on how to name them, but I've seen people referring to FQKs as _namespaced keyword for the current namespace_, which is quite long...

I guess it's best if I change the text to be more explicit.

Reply:
I've changed the text, please take a look and see if it's clear.

Pasted here for reference:

```
Always register events and subscriptions using a meaningful namespace, but don't
namespace them with `::`. We understand it's a controversial decision because
there are both pros and cons to such practice.

Whenever appropriate, it's also recommended to use _fake/invalid_ namespaces to
convey more knowledge in the keyword about which bounded context (domain) it
refers to, i.e. use dots to convey hierarchical structures.
```